cardano-testnet/src/Testnet/Process/Cli/Transaction.hs

```diff
@@ -58,25 +66,63 @@ buildTransferTx
   -> FilePath -- ^ Base directory path where the unsigned transaction file will be stored.
   -> String -- ^ Prefix for the output unsigned transaction file name. The extension will be @.txbody@.
   -> PaymentKeyInfo -- ^ Payment key pair used for paying the transaction.
-  -> PaymentKeyInfo -- ^ Payment key of the recipient of the transaction.
-  -> Int -- ^ Amount of ADA to transfer (in Lovelace).
+  -> [(TxOutAddress, Int)] -- ^ List of pairs of transaction output addresses and amounts.
   -> m (File TxBody In)
-buildTransferTx execConfig epochStateView sbe work prefix srcWallet dstWallet amount = do
-  let era = toCardanoEra sbe
-      cEra = AnyCardanoEra era
-      txBody = File (work </> prefix <> ".txbody")
-  txin <- findLargestUtxoForPaymentKey epochStateView sbe srcWallet
-  void $ execCli' execConfig
+buildTransferTx execConfig epochStateView sbe work prefix srcWallet txOutputs = do
+
+  txIn <- findLargestUtxoForPaymentKey epochStateView sbe srcWallet
+  fixedTxOuts :: [String] <- computeTxOuts
+  void $ execCli' execConfig $
     [ anyEraToString cEra, "transaction", "build"
     , "--change-address", srcAddress
-    , "--tx-in", T.unpack $ renderTxIn txin
-    , "--tx-out", destAddress <> "+" <> show amount
-    , "--out-file", unFile txBody
+    , "--tx-in", T.unpack $ renderTxIn txIn
+    ] ++ fixedTxOuts ++
+    [ "--out-file", unFile txBody
     ]
   return txBody
   where
+    era = toCardanoEra sbe
+    cEra = AnyCardanoEra era
+    txBody = File (work </> prefix <> ".txbody")
     srcAddress = T.unpack $ paymentKeyInfoAddr srcWallet
-    destAddress = T.unpack $ paymentKeyInfoAddr dstWallet
+    computeTxOuts = concat <$> sequence
+      [ case txOut of
+          PKAddress dstWallet ->
+            return ["--tx-out", T.unpack (paymentKeyInfoAddr dstWallet) <> "+" ++ show amount ]
+          ReferenceScriptAddress (File referenceScriptJSON) -> do
+            scriptAddress <- execCli' execConfig [ anyEraToString cEra, "address", "build"
+                                                 , "--payment-script-file", referenceScriptJSON
+                                                 ]
+            return [ "--tx-out", scriptAddress <> "+" ++ show amount
```

Comment:
Just to be clear, the intention is to: 
- Send ADA to a reference script address 
- Attach the reference script to an output at the reference script address

Right?

Reply:
Yes, exactly. Well, I was trying to get a reference script address somewhere on-chain but that works